src/sources/postgresql_metrics.rs

```diff
@@ -202,24 +198,26 @@ impl DatnameFilter {
             pg_stat_database_sql += &match_sql;
         }
         match (include_null, exclude_null) {
-            (false, false) => {} // nothing
-            (false, true) => {
+            // Nothing
+            (false, false) => {}
+            // Include tracking objects not in database
+            (true, false) => {
                 pg_stat_database_sql += if match_sql.is_empty() {
                     " WHERE"
                 } else {
-                    " AND"
+                    " OR"
                 };
-                pg_stat_database_sql += " datname IS NOT NULL";
+                pg_stat_database_sql += " datname IS NULL";
             }
-            (true, false) => {
+            // Exclude tracking objects not in database, precedence over include
+            (false, true) | (true, true) => {
```

Comment:
:+1: I might have suggested to just build the query with both conditions for `(true, true)` and let postgres optimize it away:

```
WHERE (`datname` ~ 'foo' OR 'datname` IS NULL) AND NOT (`datname` ~ 'foobar' OR`datname` IS NULL)
```

Where the `null` `datname`s would be excluded. I think this will result in more legible queries, but this seems ok too. I think you could do this by pushing the `null` clause addition down into `build_match_params`.

I see that what you have currently will generate clauses like:

```
WHERE (`datname` ~ 'foo') OR 'datname` IS NULL AND NOT (`datname` ~ 'foobar') AND `datname` IS NOT NULL
```

Which with Postgres's operator precedence effectively gives you:

```
WHERE (`datname` ~ 'foo') OR ('datname` IS NULL AND (NOT(`datname` ~ 'foobar') AND `datname` IS NOT NULL))
```

So it seems like it works, but relies on `AND` having a higher precedence than `OR` so it requires a bit more attention to see what's happening.

Reply:
Aha, I see, yes. What about `include_databases = ["foo", ""], exclude_databases = ["foobar"]`? Would it be:

```
WHERE (`datname` ~ 'foo') AND NOT (`datname` ~ 'foobar') OR `datname` IS NULL
```

?

Reply:
Yes, that's correct!